### PR TITLE
EES-5024 - adding App Gateway to Bicep deployment

### PR DESCRIPTION
**Pull request by duncan-at-hiveit**
:warning: The testing of this automation will require some time without an App Gateway on Dev, as the existing one is replaced.  No doubt this'll take a couple of days!  Therefore I've put the "Do No Merge" label on until we agree a window of time to do this.

This PR:
- adds modules for the Bicep App Gateway and its WAF policy.
- adds configuration to host the Public API Container Apps through it.

This PR assumes that a pre-existing certificate for the Container App will have been set up prior to standing up the App Gateway. I've added instructions to do this in the Devops Wiki on the "Rolling out Public API infrastructure" page.

### CODE REVIEW

File: infrastructure/templates/public-api/components/wafPolicy.bicep [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+    managedRules: {
+      managedRuleSets: [
+        {
+          ruleSetType: 'OWASP'
+          ruleSetVersion: '3.2'
+        }
+        {
+          ruleSetType: 'Microsoft_BotManagerRuleSet'
+          ruleSetVersion: '0.1'
```
Comment (jack-hive) [inline infrastructure/templates/public-api/components/wafPolicy.bicep:32]: should this be `1.0`?

**Review by jack-hive — CHANGES_REQUESTED** [review]

File: infrastructure/templates/public-api/components/appGateway.bicep [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+// Create a user-assigned managed identity for the App Gateway. App Gateway does not support system-assigned identities.
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: managedIdentityName
+  location: location
+}
+
+// Allow the managed identity to access certificates from Key Vault.
+module managedIdentityKeyVaultAccessPolicy 'keyVaultAccessPolicy.bicep' = {
+  name: '${appGatewayFullName}KeyVaultAccessPolicy'
```
[the remark this answers is not in the document]
Comment (duncan-at-hiveit) [reply infrastructure/templates/public-api/components/appGateway.bicep:56]: No, we don't for module names.

File: infrastructure/templates/public-api/components/wafPolicy.bicep (hunk 1 above)
Comment (duncan-at-hiveit) [reply infrastructure/templates/public-api/components/wafPolicy.bicep:32]: This is what was extracted from the templates that were set up when I created the App Gateway manually, so this is what would have been set via the UI.  I'm happy to update and try!

Updated ta - and great spot!

Comment (jack-hive) [reply infrastructure/templates/public-api/components/wafPolicy.bicep]: no worries! I had a look and it seemed that 1.0 was the latest :)

**Review by jack-hive — APPROVED** [review]

**Pull request merged**